### PR TITLE
feat: track index cache hit rate

### PR DESCRIPTION
**Pull request by rok**
See #1525

### CODE REVIEW

File: rust/lance/src/index/cache.rs [hunk 1]
```diff
@@ -19,17 +19,37 @@ use moka::sync::{Cache, ConcurrentCacheExt};
...
+
+#[derive(Debug, Default)]
+pub struct CacheStats {
+    hits: AtomicU64,
+    misses: AtomicU64,
+}
+
+impl CacheStats {
+    pub fn record_hit(&self) {
+        self.hits.fetch_add(1, Ordering::Relaxed);
```
Comment (westonpace) [inline rust/lance/src/index/cache.rs:32]: With relaxed ordering here you will get inaccurate counts, but I suppose that isn't that much of a problem for this sort of counter.

File: rust/lance/src/index/cache.rs [hunk 2]
```diff
@@ -45,6 +65,11 @@ impl IndexCache {
     }
 
     pub(crate) fn get_vector(&self, key: &str) -> Option<Arc<dyn VectorIndex>> {
+        if self.vector_cache.contains_key(key) {
```
Comment (westonpace) [inline rust/lance/src/index/cache.rs:68]: Should you also record hits/misses from `get_scalar`?

**Review by westonpace — APPROVED** [review]
This is good, are you hoping to eventually expose this through some sort of statistics or diagnostics API?

File: rust/lance/src/index/cache.rs (hunk 1 above)
Comment (rok) [reply rust/lance/src/index/cache.rs:32]: Yeah, I'd prefer not to risk index latency for potential accuracy here.

File: rust/lance/src/index/cache.rs (hunk 2 above)
Comment (rok) [reply rust/lance/src/index/cache.rs:68]: Will do!

**Comment by rok** [comment]
> This is good, are you hoping to eventually expose this through some sort of statistics or diagnostics API?

That's a fair point, I'll make another PR.

**Pull request merged**